hera_mc/cm_dataview.py

```
@@ -15,85 +15,161 @@
+        methods in geo_handling
+        """
+        print("Part number    x/E input    y/N input")
+        print("-----------    ---------    ---------")
+        for fv in fully_connected:
+            print("{:11}    {:9}    {:9}".format(fv['station_name'],
+                  fv['correlator_input_x'], fv['correlator_input_y']))
+
+    def read_db(self, parts, start_date, stop_date, dt, full_req):
+        """
+        Reads the database to produce a "map" of fully connected antennas,
+        returned in a dictionary (fc_map)
+
+        Parameters:
+        ------------
+        parts:  list of parts (stations) to search for
+        start_date:  astropy.time to start map
+        stop_date:  astropytime to stop map
+        dt:  time-step for dictionary
+        full_req:  parts needed for full Hookup
+        """
+
+        fc_map = {}
+        for p in parts:
+            print("Finding ", p)
+            fc_map[p] = {'datetime': [], 'flag': [], 'fc': []}
+            at_date = start_date
+            while at_date < stop_date:
+                fc = cm_revisions.get_full_revision(p, at_date, full_req, self.session)
+                fc_map[p]['datetime'].append(at_date.datetime)
+                fc_map[p]['flag'].append(len(fc))
+                fc_map[p]['fc'].append(fc)
+                at_date += TimeDelta(dt * 24 * 3600, format='sec')
+        self.parts_list = parts
+        self.fc_map = fc_map
+        return fc_map
+
+    def write_fc_map_file(self, filename, output='flag'):
+        """
+        Writes the fc_map to the named filename.  Either a 'flag' file can be written
+        (i.e. 0 or 1) for the plotter, or the 'corr'elator hookups
+
+        Parameters:
+        ------------
+        filename:  name of output file
+        output:  type of file 'flag' or 'corr'
+        """
+
+        if self.fc_map is None or self.parts_list is None:
+            print("You first need to generate fc_map and parts_list")
```

This should raise an exception.